### DISCUSSION TITLE
When prConcurrentLimit is set to 1, and old branch exists, renovate will not create any new PRs. Any workarounds?

### DISCUSSION THREAD

**Discussion by ocofaigh** [renovatebot/renovate#16426] [Request Help] [answered]
I have the following set `"prConcurrentLimit": 1` so that renovate only ever creates 1 PR per repo at a time. However the title of this setting is not quite accurate because this setting also means that renovate is only allowed to create 1 concurrent branch.
The problem with this is the following use case:

1. renovate creates a PR
2. PR is closed (but branch is not deleted)
3. renovate then cannot create another PR because this branch still exists. The log shows:
```
DEBUG: Calculating branchConcurrentLimit (1) (repository=GoldenEye/ocp-service-mesh-module)
DEBUG: 1 already existing branches found: renovate/terraform-module (repository=GoldenEye/ocp-service-mesh-module)
DEBUG: Branch concurrent limit remaining: 0 (repository=GoldenEye/ocp-service-mesh-module)
DEBUG: Calculated maximum branches remaining this run (repository=GoldenEye/ocp-service-mesh-module)
       "branchesRemaining": 0
```

Is there any any way to tell renovate to re-use this branch and create a PR?

**Comment by PhilipAbed** [comment, marked as answer]
renovate actually has [branchConcurrentLimit](https://docs.renovatebot.com/configuration-options/#branchconcurrentlimit)

Quoting:
> If you want the same limit for both concurrent branches and concurrent PRs, then just set a value for prConcurrentLimit and it will be reused for branch calculations too. But if you want to allow more concurrent branches than concurrent PRs, you can configure both values (e.g. branchConcurrentLimit=5 and prConcurrentLimit=3).

**Reply by ocofaigh** [reply]
Oh very good, not sure how I missed this. Will give this a try now. Thank you

**Comment by PhilipAbed** [comment]
Usually,  if you Rebase the PR it will re-use the branch, but sometimes you will be required to change the name of the PR since if you press `Close this pr` it will be considered an Ignored PR and renovate won't try to re-run it, so i suggest that if you close a PR also delete the associated branch with it then renovate will ignore it completely

**Reply by ocofaigh** [reply]
Yea that is what happens most of the time, but working on a large project with many collaborators, not everyone does this all the time. I may write some automation to delete renovate branches that do not belong to open PRs. 
PS We use `"recreateClosed": true` so may not be an issue here?

**Comment by PhilipAbed** [comment]
when you said step 2, PR closed, but branch is not deleted, did renovate close a branch automatically without deleting the branch? or did you manually closed the PR but  forgot to delete the branch?

**Reply by ocofaigh** [reply]
Manually closed the PR and branch was not deleted

**Reply by PhilipAbed** [reply]
if you do something manually i don't think renovate can do anything about it :D that's normal

**Comment by rarkins** [comment]
Use the dependency dashboard to approve the rate limited PR for creation
